File: app/src/main/java/com/ronyut/flightmobileapp/API/MyJsonObjectRequest.java
package com.ronyut.flightmobileapp.API;

import androidx.annotation.Nullable;

import com.android.volley.NetworkResponse;
import com.android.volley.ParseError;
import com.android.volley.Response;
import com.android.volley.toolbox.HttpHeaderParser;
import com.android.volley.toolbox.JsonRequest;
import java.io.UnsupportedEncodingException;
import org.json.JSONException;
import org.json.JSONObject;

public class MyJsonObjectRequest extends JsonRequest<JSONObject> {

    public MyJsonObjectRequest(
            int method,
            String url,
            @Nullable JSONObject jsonRequest,
            Response.Listener<JSONObject> listener,
            @Nullable Response.ErrorListener errorListener) {
        super(
                method,
                url,
                (jsonRequest == null) ? null : jsonRequest.toString(),
                listener,
                errorListener);
    }

    public MyJsonObjectRequest(
            String url,
            @Nullable JSONObject jsonRequest,
            Response.Listener<JSONObject> listener,
            @Nullable Response.ErrorListener errorListener) {
        this(
                jsonRequest == null ? Method.GET : Method.POST,
                url,
                jsonRequest,
                listener,
                errorListener);
    }

    @Override
    protected Response<JSONObject> parseNetworkResponse(NetworkResponse response) {
        try {
            String jsonString = new String(response.data,
                    HttpHeaderParser.parseCharset(response.headers, PROTOCOL_CHARSET));

            JSONObject result = new JSONObject();
            result.put("code", String.valueOf(response.statusCode));

            return Response.success(result,
                    HttpHeaderParser.parseCacheHeaders(response));
        } catch (Exception e) {
            return Response.error(new ParseError(e));
        }
    }
}

File: app/src/main/java/com/ronyut/flightmobileapp/api/MyJsonObjectRequest.kt
package com.ronyut.flightmobileapp.api

import com.android.volley.NetworkResponse
import com.android.volley.ParseError
import com.android.volley.Response
import com.android.volley.toolbox.HttpHeaderParser
import com.android.volley.toolbox.JsonRequest
import org.json.JSONObject

/*
    Custom JsonObjectRequest that return a json with the server's response code
    Author: Rony Utevsky
    Date: 23-06-2020
 */
class MyJsonObjectRequest(
    method: Int,
    url: String?,
    jsonRequest: JSONObject?,
    listener: Response.Listener<JSONObject?>?,
    errorListener: Response.ErrorListener?
) : JsonRequest<JSONObject>(
    method,
    url,
    jsonRequest?.toString(),
    listener,
    errorListener
) {

    override fun parseNetworkResponse(response: NetworkResponse): Response<JSONObject> {
        return try {

            val result = JSONObject()
            result.put("code", response.statusCode)
            Response.success(
                result,
                HttpHeaderParser.parseCacheHeaders(response)
            )
        } catch (e: Exception) {
            Response.error(ParseError(e))
        }
    }
}